clusterloader2/pkg/prometheus/clients/aks_managed.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prom
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	
+	"time"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+// aksManagedPrometheusClient talks to the Azure Managed Prometheus.
+// This only works if the cluster is hosted in AKS.
+// Overview: https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/prometheus-metrics-overview.
+// Query API: https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/prometheus-api-promql .
+type aksManagedPrometheusClient struct {
+	uri   			string
+	headers    		map[string]string
+}
+
+func (mpc *aksManagedPrometheusClient) Query(query string, queryTime time.Time) ([]byte, error) {
+	req, err := http.NewRequest("GET", mpc.uri, nil)
+    if err != nil {
+        return nil, err
+    }
+
+	// Set query parameters
```

Comment:
Seems like something weird is going on with indentation. Are you using all tabs?